padertorch/contrib/examples/speaker_classification/supervised/train.py

```diff
@@ -0,0 +1,119 @@
+"""
+Example call:
+
+export STORAGE_ROOT=<your desired storage root>
+python -m padertorch.contrib.examples.speaker_classification.supervised.train with database_json=</path/to/json>
+"""
+import os
+from pathlib import Path
+
+from torch.nn import GRU
+from sacred import Experiment, commands
+
+from padertorch.io import get_new_storage_dir
+from padertorch import Trainer
+from padertorch.contrib.examples.speaker_classification.supervised.model import SpeakerClf
+from padertorch.contrib.je.modules.conv import CNN1d
+from padertorch.modules.fully_connected import fully_connected_stack
+from padertorch.train.optimizer import Adam
+from padertorch.modules.normalization import Normalization
+from padertorch.contrib.examples.speaker_classification.supervised.data import get_datasets
+
+
+ex = Experiment('speaker_clf')
+
+
+@ex.config
+def defaults():
+    database_json = (
+        Path(os.environ['NT_DATABASE_JSONS_DIR']) / 'librispeech.json'
+        if 'NT_DATABASE_JSONS_DIR' in os.environ else None
+    )
+    assert database_json is not None, (
+        'database_json cannot be None.\n'
+        'Either start the training with "python -m padertorch.contrib.examples.'
+        'speaker_classification.train with database_json=</path/to/json>" '
+        'or export "NT_DATABASE_JSONS_DIR" which points to a directory with a '
+        '"librispeech.json" prior to training start (see README for the '
+        'JSON format).'
+    )
+    dataset = 'train_clean_100'
+    batch_size = 16
+    num_speakers = 251
+
+
+def get_model(num_speakers):
+    feature_extractor = Normalization(
+        data_format='bft',
+        shape=(None, 64, None),
+        statistics_axis='bt',
+        independent_axis=None,
+    )
+    cnn = CNN1d(
+        in_channels=64,
+        out_channels=4*[512],
+        output_layer=False,
+        kernel_size=5,
+        norm='batch'
+    )
+    gru = GRU(input_size=512, hidden_size=256, num_layers=2, batch_first=True)
+    fcn = fully_connected_stack(
+        256, hidden_size=[256], output_size=num_speakers, dropout=0.
+    )
+
+    speaker_clf = SpeakerClf(feature_extractor, cnn, gru, fcn)
+    return speaker_clf
+
```

Comment:
Why not move this into the config? Since that would be more similar to our real workflow

Reply:
I guess I could do that. I kept this from the previous version were sacred was not yet used and missed to move that part. I'll change that.

Reply:
Updated in commit https://github.com/fgnt/padertorch/pull/57/commits/697cb3eb65d506f4df19ee730135936b52cd5f92

Reply:
Here class_to_str is not necessary, if you call SpeakerClf.get_config(model). 
This should be done because otherwise all values not specified in this config are not written to the config.json. This might be a problem in case of changing defaults.

You could think about moving the options from the trainer to the config as well.

Reply:
You could call `pt.Configurable.get_config(model)` after you defined the model, that would fill the config with all defaults.
i.e. the config would then be more verbose.

I think he calls `class_to_str`, because he uses sacred to save the config.
You could use `pt.configurable.recursive_class_to_str` before you write the config, then the values would also be str, when you write the config.

Reply:
Should now be considered by commit https://github.com/fgnt/padertorch/pull/57/commits/7f9eaaf7c740f13092c3fad36cfb457c93bbeafb